### DISCUSSION TITLE
How does the `na.action` parameter work?

### DISCUSSION THREAD

**Discussion by ringo-alin** [kogalur/randomForestSRC#203] [Q&A] [answered]
Hi, I used to think that the `rfrsc.grow` would grow on a complete dataset when `na.action = 'na.impute'` (step 1, data imputation, step 2, forest growing). But the following test  gives different results.
```
data(pbc, package = "survival")
pbc.cr <- rfsrc(Surv(time, status) ~ ., pbc, na.action = "na.impute", seed = 1)
print(pbc.cr)
new.pbc <- pbc
new.pbc[pbc.cr$imputed.indv, names(pbc.cr$imputed.data)] <- pbc.cr$imputed.data[, names(pbc.cr$imputed.data)]
rfsrc(Surv(time, status) ~ ., new.pbc, seed =1)
```
So, I wonder
1) how to get the complete data used to train the model? (I opened an issue #105 before, but I didn't get a solution there)
2) will competing risk settings, such as `cause`, affect data imputation? (It doesn't take effects in `impute` function)

**Comment by ishwaran** [comment, marked as answer]
You should pre-impute the data.   Then using the imputed data, run a RSF competing risk analysis - now there is no need to deal with `na.action`

Here's how using the missForest imputation method (without using the outcomes):

```
## read data
data(pbc, package = "survival")

## separate x features from survival outcomes
pbc.x <- pbc[, !(names(pbc) %in% c("time", "status"))]
pbc.y <- pbc[, names(pbc) %in% c("time", "status")]

## use missForest to impute x-features
pbc.x.imp <- impute(data = pbc.x, mf.q = 1)

## assemble the imputed data set
pbc.imp <- data.frame(pbc.y, pbc.x.imp)

## run RSF competing risks on imputed data
pbc.cr <- rfsrc(Surv(time, status) ~ ., pbc.imp)
print(pbc.cr)



```
